OptimizelySDKShared/OptimizelySDKShared/OPTLYDatafileConfig.m

```diff
@@ -17,33 +17,65 @@
 #import "OPTLYDatafileConfig.h"
 #import "OPTLYManagerBase.h"
 
-NSString * const OPTLY_DATAFILE_URL = @"https://cdn.optimizely.com/json/%@.json";
+NSString * const DEFAULT_HOST = @"https://cdn.optimizely.com";
+NSString * const OPTLY_PROJECTID_SUFFIX = @"/json/%@.json";
+NSString * const OPTLY_ENVIRONMENTS_SUFFIX = @"/datafiles/%@.json";
 
 @interface OPTLYDatafileConfig()
+- (nullable id)initWithProjectId:(NSString *)projectId withSDKKey:(NSString *)sdkKey withHost:(NSString *)host;
 @property(nonatomic, strong, nullable) NSString* projectId;
 @property(nonatomic, strong, nullable) NSString* sdkKey;
+@property(nonatomic, strong) NSString* host;
+@property(nonatomic, strong) NSURL* projectIdUrl;
+@property(nonatomic, strong) NSURL* sdkKeyUrl;
+@end
+
+@implementation OPTLYDatafileConfig(OPTLYHelpers)
++ (NSString *)defaultProjectIdPath:(NSString *)projectId {
+    NSString *datafileFormat = [DEFAULT_HOST stringByAppendingString:OPTLY_PROJECTID_SUFFIX];
+    return [NSString stringWithFormat:datafileFormat, projectId];
+}
++ (NSString *)defaultSdkKeyPath:(NSString *)sdkKey {
+    NSString *datafileFormat = [DEFAULT_HOST stringByAppendingString:OPTLY_ENVIRONMENTS_SUFFIX];
+    return [NSString stringWithFormat:datafileFormat, sdkKey];
+}
 @end
 
 @implementation OPTLYDatafileConfig
 
-- (instancetype)initWithProjectId:(NSString *)projectId withSDKKey:(NSString *)sdkKey {
+- (instancetype)initWithProjectId:(NSString *)projectId withSDKKey:(NSString *)sdkKey withHost:(NSString *)host {
+    self.host = host;
+    
     if (![OPTLYManagerBase isValidKeyString:projectId] && ![OPTLYManagerBase isValidKeyString:sdkKey]) {
         // One of projectId and sdkKey needs to be a valid key string.
         return nil;
     }
     if (self = [super init]) {
         self.projectId = projectId;
         self.sdkKey = sdkKey;
+        if (self.projectId != nil) {
+            NSString *datafileFormat = [self.host stringByAppendingString:OPTLY_PROJECTID_SUFFIX];
+            NSString *datafile = [NSString stringWithFormat:datafileFormat, self.projectId];
+            self.projectIdUrl = [NSURL URLWithString:datafile];
+        }
+        if (self.sdkKey != nil) {
+            NSString *datafileFormat = [self.host stringByAppendingString:OPTLY_ENVIRONMENTS_SUFFIX];
+            NSString *datafile = [NSString stringWithFormat:datafileFormat, self.sdkKey];
+            self.sdkKeyUrl = [NSURL URLWithString:datafile];
```

Comment:
Same comment as in the Android PR, do we need 2 separate URL properties? We can just have 1 that gets set on init and use that one throughout the code